### ISSUE TITLE
Add a link to "edit this page"

### ISSUE THREAD

**Issue by betatim** [jupyterhub/alabaster-jupyterhub#12]
How hard is it to add a link (to the top or bottom) or other prominent/easy to find location that takes you straight to the source in "edit mode" on GitHub for the page of the docs that I am viewing?

Inn several projects I kinda know where the source is but when I see a small typo or something I still put it off "till later" because I am not quite sure. My guess is for other people less familiar with the code base it is also hard to find the source of each page (quick fire quiz round: which repo hosts docs.mybinder.org? (I don't know)). The end result is putting off edits till later. If someone knows the sphinx magic needed (or extension that already exists) to take us straight to the editable source that would be a great improvement.

(If someone can guide me I'll make a PR)

**Comment by choldgraf** [comment]
I wonder if @kirstiejane could help with this? I think she worked on something similar recently...

I agree this would be great

**Comment by parente** [comment]
FWIW, https://docs.readthedocs.io/en/stable/guides/remove-edit-buttons.html seems to suggest RTD adds the buttons automatically, maybe to certain themes. I don't see them in the header of the pure-alabaster based https://jupyter-docker-stacks.readthedocs.io/en/latest/, but it does show up in the popup menu:

![Screen Shot 2019-04-23 at 11 19 31 AM](https://user-images.githubusercontent.com/153745/56593068-c598ee80-65b9-11e9-9380-2ec0a5432936.png)

https://jupyterlab.readthedocs.io/en/stable/ and https://jupyter-notebook.readthedocs.io/en/stable/ both have them in the top right instead of in the RTD menu.

**Comment by betatim** [comment]
Maybe we can find how RTD inserts them. I think the RTD menu thing (the black popup in your screenshot) is missing from pages using this theme because it is its own theme and not a RTD theme :-/

**Comment by choldgraf** [comment]
I think we could add this by riffing off of the alabster_jupyterhub navigation HTML:

https://github.com/bitprophet/alabaster/blob/master/alabaster/navigation.html

and use the username/repo variables in the html_config dictionary:

https://alabaster.readthedocs.io/en/latest/customization.html#theme-options

the trick, then, would be to figure out how the proper links to the page rst paths using the current page location.

**Comment by choldgraf** [comment]
another option would be to actually piggy-back off of the RTD popout. The link for GitHub "edit this page" is under a `dd` tag with the text `Edit`, so we could add a little bit of javascript in the sidebar that basically does:

```javascript
var popOutLinks = document.querySelectorAll('dd');
for (ii=0; ii < popOutLinks.length; ii++) {
    var thisPopOut = popOutLinks[ii];
    if (thisPopOut.textContent == "Edit") {
        // grab the href and add it to the sidebar
    }
} 
```

**Comment by choldgraf** [comment]
It looks like this is where the RTD theme adds this:

https://github.com/rtfd/sphinx_rtd_theme/blob/ceb2ef463cf2aa8891ce97f0cbae854a0a9f37a5/sphinx_rtd_theme/breadcrumbs.html#L45

**Comment by choldgraf** [comment]
another useful lead, this is the base HTML that sphinx uses to generate the "go to source" link:

https://github.com/sphinx-doc/sphinx/blob/master/sphinx/themes/basic/sourcelink.html

that gets us pretty close to what we want, we just need that to be the "edit" mode of the source, instead of the source page itself

**Comment by KirstieJane** [comment]
I folks! Thank you for tagging me @choldgraf.....but unfortunately the idea I had was a dream and not something that I actually know how to to!

I'm following the conversation eagerly though - happy to test drive anything if that's useful (but unfortunately it may be the limit of my ability in this case 😭)

**Comment by KirstieJane** [comment]
Here's a really simple example of where I've seen this done though:

https://mozilla.github.io/open-leadership-training-series/articles/introduction-to-open-leadership/

The `suggest changes` button takes you to the markdown file in github. Here's a link to the line in the jekyll setup: https://github.com/mozilla/open-leadership-training-series/blob/ac24c3084f5fa9bc18b62dd6c5a7033463540951/_layouts/page.html#L23

**Comment by betatim** [comment]
We now have the RTD popup area thing in the bottom right corner for https://binderhub.readthedocs.io/en/latest/ which means I am happy.

**Comment by choldgraf** [comment]
Wait I'm a bit confused, that's always been there...

**Comment by betatim** [comment]
What? In that case I need to get my eyes checked 👀 !

An even easier to find link would be great and would help people like me who can't read :-/

**Comment by choldgraf** [comment]
yeah, I don't think that little box is very discoverable, I'd prefer a link that's on the same page and immediately visible

**Comment by statiksof** [comment]
@choldgraf I am using your customized theme for our documentation. Is there a way to do the opposite? I mean removing "Edit this page". From the offcial RTD documentation, they describe a workaround by extending a template file. What would be that file in your theme?
